### PR TITLE
Adding migration guide for dashboard and fixing signing doc

### LINKED ISSUE #45895: Docs: Update plugin migration guide in regards to Grafana v8 dashboard includes with addToNav set to true
[Plugin migration from Grafana v7 to v8](https://grafana.com/docs/grafana/latest/developers/plugins/migration-guide/#from-version-7xx-to-8xx) needs an update with a section for dashboard includes that has [addToNav](https://grafana.com/docs/grafana/latest/developers/plugins/metadata/#properties-4) set to true. When this is true a [uid](https://grafana.com/docs/grafana/latest/developers/plugins/metadata/#properties-4) needs to be set to the same `uid` set in the `<dashboard>.json`. Without these changes navigating from Grafana menu directly to an included dashboard will fail.

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What this PR does / why we need it**:
- Update documentation on migrating app plugins to grafana 8
- Fix documentation around plugin signing
 
**Which issue(s) this PR fixes**:

Fixes #45895

### CODE REVIEW

File: docs/sources/developers/plugins/migration-guide.md [hunk 1]
```diff
@@ -308,6 +319,32 @@ import { cx, css } from 'emotion';
...
+
+dashboards/example-dashboard.json
+
+{
+  // excerpt of dashboard content
+  "title": "Example Dashboard",
+  "uid": "l3KqBxCMz",
+  "version": 1
+}
+```
```
Comment (leventebalogh) [inline docs/sources/developers/plugins/migration-guide.md:346]: **Nit:** maybe formatting like this instead (to highlight the changes)?

**`plugin.json`**
```diff
{
  "type": "dashboard",
  "name": "(Team) Situation Overview",
  "path": "dashboards/example-dashboard.json",
  "addToNav": true,
  "defaultNav": false,
+  "uid": "l3KqBxCMz"
}
```

**`dashboards/example-dashboard.json`**
```diff
{
  "title": "Example Dashboard",
+  "uid": "l3KqBxCMz",
  "version": 1
  // ...
}
```
```

**Review by leventebalogh — APPROVED** [review]
LGTM 🚀

**Review by jackw — APPROVED** [review]
LGTM! 👏

File: docs/sources/developers/plugins/migration-guide.md (hunk 1 above)
Comment (leventebalogh) [reply docs/sources/developers/plugins/migration-guide.md:346]: Sorry, probably this won't really work in [Hugo](https://gohugo.io/) as I imagined, just ignore my comment.

Comment (marefr) [inline docs/sources/developers/plugins/migration-guide.md]: Maybe I misunderstand the intent/idea, but I suggests something like this:

````suggestion
**plugin.json:**
```json
{
  "id": "plugin-id",
  // ...
  "includes": [
    {
      "type": "dashboard",
      "name": "(Team) Situation Overview",
      "path": "dashboards/example-dashboard.json",
      "addToNav": true,
      "defaultNav": false,
      "uid": "l3KqBxCMz"
    }
  ]
  // ...
}
```

**dashboards/example-dashboard.json:**
```json
{
  // excerpt of dashboard content
  "title": "Example Dashboard",
  "uid": "l3KqBxCMz",
  "version": 1
}
```
````

File: docs/sources/developers/plugins/migration-guide.md [hunk 2]
```diff
@@ -308,6 +319,32 @@ import { cx, css } from 'emotion';
 import { cx, css } from '@emotion/css';
 ```
 
+### Update needed for app plugins using dashboards
+
+To make side navigation work properly - app plugins targeting Grafana 8.+ and using navigation bar need to adjust their `plugin.json` and all dashboard json files to have a matching `uid`.
```
Comment (marefr) [inline docs/sources/developers/plugins/migration-guide.md:324]: Don't have time now but maybe mention app including dashboard with a link to it in the navigation bar using `addToNav` property.

**Review by marefr — CHANGES_REQUESTED** [review]
Great work! Please have a look at my comment/suggestion.

File: docs/sources/developers/plugins/migration-guide.md (hunk 1 above)
Comment (tolzhabayev) [reply docs/sources/developers/plugins/migration-guide.md]: @marefr good point, changed the example to match the structure of plugin.json better.

File: docs/sources/developers/plugins/migration-guide.md (hunk 2 above)
Comment (tolzhabayev) [reply docs/sources/developers/plugins/migration-guide.md]: @marefr added the addToNav part and link to it's definition

**Review by marefr — APPROVED** [review]
LGTM

**Review by marcusolsson — APPROVED** [review]

**Pull request merged**